### PR TITLE
refactor(flows): replace cmsSlug with flowType

### PR DESCRIPTION
**Pull request by chohner**
- replaces `cmsSlug` with `flowType`
- replaces hard-coded progress with filtering by `flowType`

### CODE REVIEW

**Review by frederike-ramin — APPROVED** [review]
Love it.

File: app/flows/flows.server.ts [hunk 1]
```diff
@@ -19,8 +18,11 @@ export type FlowMigration = {
   buttonUrl?: string;
 };
 
+const flowTypes = ["vorabCheck", "formFlow"] as const;
```
Comment (aaschlote) [inline app/flows/flows.server.ts:21]: shouldn't it be only `vorabcheck` without capital `C`? I believe it's only one word, right?

File: app/flows/flows.server.ts [hunk 2]
```diff
@@ -19,8 +18,11 @@ export type FlowMigration = {
...
+export type FlowType = (typeof flowTypes)[number];
```
Comment (aaschlote) [inline app/flows/flows.server.ts:22]: doing only:
```
export type FlowType = "vorabcheck" | "formFlow"
```

is it not enough? Or do we have other problems?

Comment (chohner) [reply app/flows/flows.server.ts]: I just forgot to push

File: app/services/cms/apiFromFlowType.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import type { FlowType } from "~/domains/flows.server";
+import type { FlowPageId } from "./schemas";
+
+const apiMapping = {
+  formFlow: "form-flow-pages",
+  vorabCheck: "vorab-check-pages",
+} satisfies Record<FlowType, FlowPageId>;
```
Comment (pgurusinga) [inline app/services/cms/apiFromFlowType.ts:7]: Why `result-pages` is not part of the mapping?

Comment (chohner) [reply app/services/cms/apiFromFlowType.ts:7]: There is no `FlowType: result-page`

**Pull request merged**